include/GafferSceneUI/ContextAlgo.h

```diff
@@ -74,6 +74,10 @@ void expand( Gaffer::Context *context, const GafferScene::PathMatcher &paths, bo
 /// Returns a new PathMatcher containing the new leafs of this expansion.
 GafferScene::PathMatcher expandDescendants( Gaffer::Context *context, const GafferScene::PathMatcher &paths, const GafferScene::ScenePlug *scene, int depth = Imath::limits<int>::max() );
 
+/// Removes paths from the current expansion.
+/// Returns a new PathMatcher containing the removed paths.
+GafferScene::PathMatcher collapse( Gaffer::Context *context, const GafferScene::PathMatcher &paths );
```

Comment:
I also got a crash when collapsing the selection that I was unable to reproduce subsequently, so I'm inclined to ditch this change until we have another use case to inform what the right semantics might be, and we can focus on it alone.